src/encoded/upgrade/user.py

```diff
@@ -42,3 +42,10 @@ def user_7_8(value, system):
             value['viewing_groups'].append('community')
     else:
         value['viewing_groups'] = ['community']
+
+
+@upgrade_step('user', '8', '9')
+def user_8_9(value, system):
+    if 'groups' in value:
+        new_groups = [group for group in value['groups'] if group in ['admin', 'verified', 'read-only-admin']]
+        value['groups'] = new_groups
```

Comment:
why not collapse into one line? value['groups'] = [group for group in value['groups']......] ?